src/components/TimePicker/__test__/timePicker.spec.js

```diff
@@ -1,34 +1,139 @@
     });
     it('should fire onFocus with the value passed', () => {
         const onFocusMockFn = jest.fn();
-        const component = mount(<TimePicker value="18:35" onFocus={onFocusMockFn} />);
+        const component = mount(
+            <TimePicker label="unit-testing-timePicker" value="18:35" onFocus={onFocusMockFn} />,
+        );
         component.find('input').simulate('focus');
         component.find('input').simulate('blur');
         expect(onFocusMockFn).toHaveBeenCalledWith('18:35');
     });
+    it('should set isOpen to true and fire onClick when readOnly is not passed', () => {
+        const onClickMockFn = jest.fn();
+        const component = mount(
+            <TimePicker label="unit-testing-timePicker" value="18:35" onClick={onClickMockFn} />,
+        );
+        component.find('input').simulate('click');
+        expect(onClickMockFn).toHaveBeenCalledTimes(1);
+        expect(component.find('Modal').prop('isOpen')).toBe(true);
+    });
+    it('should not set isOpen to true and not fire onClick when readOnly is passed', () => {
+        const onClickMockFn = jest.fn();
+        const component = mount(
+            <TimePicker
+                readOnly
+                label="unit-testing-timePicker"
+                value="18:35"
+                onClick={onClickMockFn}
+            />,
+        );
+        component.find('input').simulate('click');
+        expect(onClickMockFn).toHaveBeenCalledTimes(0);
+        expect(component.find('Modal').prop('isOpen')).toBe(false);
+    });
+    it('should have the right class names when a custom className is passed', () => {
+        const component = mount(
+            <TimePicker label="unit-testing-timePicker" className="my-custom-class-name" />,
+        );
+        expect(
+            component.find('div.rainbow-time-picker_container.my-custom-class-name').exists(),
+        ).toBe(true);
+    });
+    it('should focus the input when the focus method is called', () => {
+        const component = mount(
+            <TimePicker label="unit-testing-timePicker" placeholder="testing-timePicker" />,
+        );
+
+        component.instance().focus();
+        const focusedElementPlaceholder = document.activeElement.getAttribute('placeholder');
+        expect(focusedElementPlaceholder).toBe('testing-timePicker');
+    });
+    it('should blur the input when the blur method is called', () => {
+        const component = mount(
+            <TimePicker label="unit-testing-timePicker" placeholder="testing-timePicker" />,
+        );
+        const instance = component.instance();
+        const focusedElementPlaceholder = document.activeElement.getAttribute('placeholder');
+
+        instance.focus();
+        expect(document.activeElement.getAttribute('placeholder')).toBe(focusedElementPlaceholder);
+        instance.blur();
+        expect(document.activeElement.getAttribute('placeholder')).toBeNull();
+    });
```

Comment:
there is a toBeFocusable custom jest matcher for not repeat this in all focusable components so remove this two test
make this test in a new file called ```timePicker.focusable.spec.js```
look at button or input component to see how it is used
this matcher check that onClick, onFocus, and onBlur events are called and that the component have the imperative methods (click, focus, blur).
in the case of the onBlur and onFocus already testes we need to keep them since what is tested is that they are called with the value and not with the native event